modules/ppcp-onboarding/src/Render/OnboardingRenderer.php

```diff
@@ -83,9 +94,17 @@ public function get_signup_link( bool $is_production, array $products ) {
 			->with_products( $products )
 			->data();
 
+		$environment = $is_production ? 'production' : 'sandbox';
+		$product     = isset( $data['products']['PPCP'] ) ? 'ppcp' : 'express-checkout';
+		if ( $this->cache->has( $environment . '-' . $product ) ) {
+			return $this->cache->get( $environment . '-' . $product );
+		}
+
 		$url = $is_production ? $this->production_partner_referrals->signup_link( $data ) : $this->sandbox_partner_referrals->signup_link( $data );
 		$url = add_query_arg( $args, $url );
 
+		$this->cache->set( $environment . '-' . $product, $url );
```

Comment:
It seems like `Cache` sets `0` for transient expiration. May cause problems if the link stops working after some time?